src/txn/txn.c

```diff
@@ -1835,7 +1835,11 @@ __wt_txn_commit(WT_SESSION_IMPL *session, const char *cfg[])
                 if (cache->hs_fileid != 0 && op->btree->id == cache->hs_fileid)
                     break;
 
-                __wt_txn_op_set_timestamp(session, op);
+                /*
```

Comment:
How about something like this `Apply the first commit timestamp instead of the current commit timestamp for the the updates that are done before the first commit timestamp is set`.